lib/web/apiserver.go

```diff
@@ -3122,9 +3122,21 @@ func (h *Handler) createSSHCert(w http.ResponseWriter, r *http.Request, p httpro
 	} = authClient
 
 	authReq := auth.AuthenticateUserRequest{
-		Username:       req.User,
-		PublicKey:      req.PubKey,
-		ClientMetadata: clientMetaFromReq(r),
+		HeadlessAuthenticationID: req.HeadlessAuthenticationID,
+		Username:                 req.User,
+		PublicKey:                req.PubKey,
+		ClientMetadata:           clientMetaFromReq(r),
+	}
+
+	if authReq.HeadlessAuthenticationID != "" {
+		// create a new http client with a standard callback timeout.
```

Comment:
Why? What's wrong with using the existing client? Mind explaining that in the comment as well?

Reply:
Updated the comment